app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/ChatFeedRecyclerAdapter.java

```diff
@@ -448,21 +447,36 @@ public boolean onResourceReady(GlideDrawable resource, String model,
                 mapViewHolder.mapImage.setOnClickListener(new View.OnClickListener() {
                     @Override
                     public void onClick(View v) {
-                        mapViewHolder.onClick(v);
                         /*
                           Open in Google Maps if installed, otherwise open browser.
                         */
-
-                        if (AndroidHelper.isGoogleMapsInstalled(currContext) && mapHelper.isParseSuccessful()) {
-                            Uri gmmIntentUri = Uri.parse(String.format("geo:%s,%s?z=%s", mapHelper.getLatitude(), mapHelper.getLongitude(), mapHelper.getZoom()));
-                            Intent mapIntent = new Intent(Intent.ACTION_VIEW, gmmIntentUri);
-                            mapIntent.setPackage(AndroidHelper.GOOGLE_MAPS_PKG);
-                            currContext.startActivity(mapIntent);
+                        if(selectedItems.size() == 0) {
+                            if (AndroidHelper.isGoogleMapsInstalled(currContext) && mapHelper.isParseSuccessful()) {
+                                Uri gmmIntentUri = Uri.parse(String.format("geo:%s,%s?z=%s", mapHelper.getLatitude(), mapHelper.getLongitude(), mapHelper.getZoom()));
+                                Intent mapIntent = new Intent(Intent.ACTION_VIEW, gmmIntentUri);
+                                mapIntent.setPackage(AndroidHelper.GOOGLE_MAPS_PKG);
+                                currContext.startActivity(mapIntent);
+                            } else {
+                                Intent mapIntent = new Intent(Intent.ACTION_VIEW);
+                                mapIntent.setData(Uri.parse(mapHelper.getWebLink()));
+                                currContext.startActivity(mapIntent);
```

Comment:
Move this out of the if block as this is common for both the if and the else statements